kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -468,7 +468,14 @@ else if (newFiles.isEmpty()) {
                 new HashSet<>(newCheckpoint.getCorrespondingFiles(logPath));
             final List<FileStatus> newCheckpointFileList = checkpoints
                 .stream()
-                .filter(f -> newCheckpointPaths.contains(new Path(f.getPath())))
+                .filter(f -> {
+                    // TODO: this is a temporary workaround until #1956 is fixed
+                    String fPath = f.getPath();
```

Comment:
Clearly this is an insufficient solution

https://github.com/delta-io/delta/actions/runs/5788126150/job/15686453489?pr=1939

```
class io.delta.kernel.internal.snapshot.SnapshotManager :: newFiles: [00000000000000000010.checkpoint.parquet, 00000000000000000010.json, 00000000000000000011.json, 00000000000000000012.json, 00000000000000000013.json, 00000000000000000014.json]
class io.delta.kernel.internal.snapshot.SnapshotManager :: 
checkpoints: [00000000000000000010.checkpoint.parquet]
deltas: [00000000000000000010.json, 00000000000000000011.json, 00000000000000000012.json, 00000000000000000013.json, 00000000000000000014.json]
class io.delta.kernel.internal.snapshot.SnapshotManager :: lastCheckpoint: CheckpointInstance{version=-1, numParts=Optional.empty}
class io.delta.kernel.internal.snapshot.SnapshotManager :: checkpointFiles: [CheckpointInstance{version=10, numParts=Optional.empty}]
class io.delta.kernel.internal.snapshot.SnapshotManager :: newCheckpointOpt: Optional[CheckpointInstance{version=10, numParts=Optional.empty}]
class io.delta.kernel.internal.snapshot.SnapshotManager :: newCheckpointVersion: 10
class io.delta.kernel.internal.snapshot.SnapshotManager :: deltasAfterCheckpoint: [00000000000000000011.json, 00000000000000000012.json, 00000000000000000013.json, 00000000000000000014.json]
class io.delta.kernel.internal.snapshot.SnapshotManager :: deltaVersions: [11, 12, 13, 14]
[error] Test io.delta.kernel.integration.TestDeltaTableReads.tableWithCheckpoint failed: java.lang.IllegalStateException: Seems like the checkpoint is corrupted. Failed in getting the file information for:
[error] [file:/home/runner/work/delta/delta/kernel/kernel-default/target/test-classes/basic-with-checkpoint/_delta_log/00000000000000000010.checkpoint.parquet]
[error] among
[error] file:/home/runner/work/delta/delta/kernel/kernel-default/target/test-classes/basic-with-checkpoint/_delta_log/00000000000000000010.checkpoint.parquet, took 0.012 sec
[error]     at io.delta.kernel.internal.snapshot.SnapshotManager.lambda$getLogSegmentForVersion$26(SnapshotManager.java:492)
[error]     at java.util.Optional.map(Optional.java:215)
[error]     at io.delta.kernel.internal.snapshot.SnapshotManager.getLogSegmentForVersion(SnapshotManager.java:466)
[error]     at io.delta.kernel.internal.snapshot.SnapshotManager.getLogSegmentForVersion(SnapshotManager.java:270)
[error]     at io.delta.kernel.internal.snapshot.SnapshotManager.getLogSegmentFrom(SnapshotManager.java:234)
[error]     at io.delta.kernel.internal.snapshot.SnapshotManager.getSnapshotAtInit(SnapshotManager.java:190)
[error]     at io.delta.kernel.internal.snapshot.SnapshotManager.buildLatestSnapshot(SnapshotManager.java:65)
[error]     at io.delta.kernel.internal.TableImpl.getLatestSnapshot(TableImpl.java:50)
[error]     at io.delta.kernel.integration.BaseIntegration.snapshot(BaseIntegration.java:74)
[error]     at io.delta.kernel.integration.TestDeltaTableReads.tableWithCheckpoint(TestDeltaTableReads.java:309)
[error]     ...
```